File: ios/Counter.m
//
//  Counter.m
//  NativePOC
//
//  Created by Satyam on 29/12/22.
//

#import <Foundation/Foundation.h>
#import <Foundation/NSDictionary.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(Counter,NSObject)

RCT_EXTERN_METHOD(increment:
                  (NSDictionary)givenValue
                  callback:(RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(decrement:
                  (RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

@end

File: ios/Counter.swift
//
//  Counter.swift
//  NativePOC
//
//  Created by Satyam on 29/12/22.
//

import Foundation


@objc(Counter)
class Counter : NSObject{
  private var count = 0;

  @objc
  func increment(_ givenValue:NSMutableDictionary, callback:RCTResponseSenderBlock){
//    count=;
    givenValue.setValue("Ajay", forKey: "s");
//    print(count);
    callback([givenValue])
  }

@objc
  static func requiresMainQueueSetup() ->Bool{
    return true;
  }
  
  @objc
  func constantsToExport()-> [String:Any]!{
    return ["initialCount":0];
  }
  
  @objc
  func decrement(_ resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock){
    if(count==0)
    {
      let error = NSError(domain: "", code: 200, userInfo: nil);
      reject("ERROR_COUNT","count can't be negative",error);
    }
    else
    {
      count-=1;
      resolve(count);
    }
  }
}
